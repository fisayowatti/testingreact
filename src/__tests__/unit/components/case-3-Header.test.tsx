import React from 'react'
import { Provider as StoreProvider } from 'react-redux'
import { render, fireEvent } from '@testing-library/react'

import Header from '../../../components/Header'
import { FiltersContext } from '../../../context/filters'
import { MemoryRouter } from 'react-router-dom'
import { createStore } from '../../../store'

describe('The Header component', () => {
  const defaultContext = {
    toggleShowingFilters: jest.fn(),
  }

  const setupHeader = (
    value = defaultContext,
    routerProps = {
      initialEntries: ['/', '/products/1'],
      initialIndex: 0,
    },
  ) =>
    render(
      <FiltersContext.Provider value={...value as any}>
        <StoreProvider store={createStore()}>
          <MemoryRouter {...routerProps}>
            <Header />
          </MemoryRouter>
        </StoreProvider>
      </FiltersContext.Provider>,
    )
  it('renders header correctly', () => {
    const { asFragment } = setupHeader()

    expect(asFragment()).toMatchSnapshot()
  })

  it('toggles the filter open when the Filter button is clicked', () => {
    const { getByText } = setupHeader()

    const filterButton = getByText(/filter/i)

    fireEvent.click(filterButton)

    expect(defaultContext.toggleShowingFilters).toHaveBeenCalled()
  })

  it('shows the filter button only on the home page', () => {
    const { queryByText } = setupHeader(undefined, {
      initialEntries: ['/', '/products/1'],
      initialIndex: 1,
    })

    expect(queryByText(/filter/i)).toBeNull()
  })
})
