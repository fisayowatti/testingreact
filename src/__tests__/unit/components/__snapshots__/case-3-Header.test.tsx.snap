// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders header correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-hKgILt cMbpqS"
  >
    <span
      class="sc-dlfnbm ggyzkx"
    />
    <div
      class="sc-bdfBwQ efJNDJ"
    >
      TESTING 
      <span />
      REACT.DEV
    </div>
    <button
      class="sc-gsTCUz feUypG"
      data-testid="FilterButton"
    >
      Filter
    </button>
  </header>
</DocumentFragment>
`;
